easybuild/main.py

```diff
@@ -586,9 +481,198 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
+        close_pr(options.close_pr, motivation_msg=options.close_pr_msg)
+
+    elif options.list_prs:
+        print(list_prs(options.list_prs))
+
+    elif options.merge_pr:
+        merge_pr(options.merge_pr)
+
+    elif options.review_pr:
+        print(review_pr(pr=options.review_pr, colored=use_color(options.color), testing=testing,
+                        max_ecs=options.review_pr_max, filter_ecs=options.review_pr_filter))
+
+    elif options.add_pr_labels:
+        add_pr_labels(options.add_pr_labels)
+
+    elif options.list_installed_software:
+        detailed = options.list_installed_software == 'detailed'
+        print(list_software(output_format=options.output_format, detailed=detailed, only_installed=True))
+
+    elif options.list_software:
+        print(list_software(output_format=options.output_format, detailed=options.list_software == 'detailed'))
+
+    elif options.create_index:
+        print_msg("Creating index for %s..." % options.create_index, prefix=False)
+        index_fp = dump_index(options.create_index, max_age_sec=options.index_max_age)
+        index = load_index(options.create_index)
+        print_msg("Index created at %s (%d files)" % (index_fp, len(index)), prefix=False)
+
+    # non-verbose cleanup after handling GitHub integration stuff or printing terse info
+    early_stop_options = [
+        options.add_pr_labels,
+        options.check_eb_deps,
+        options.check_github,
+        options.create_index,
+        options.install_github_token,
+        options.list_installed_software,
+        options.list_software,
+        options.close_pr,
+        options.list_prs,
+        options.merge_pr,
+        options.review_pr,
+        options.terse,
+        search_query,
+    ]
+    if any(early_stop_options):
+        clean_exit(logfile, eb_tmpdir, testing, silent=True)
+
+    # update session state
+    eb_config = eb_go.generate_cmd_line(add_default=True)
+    modlist = modtool.list()  # build options must be initialized first before 'module list' works
+    init_session_state.update({'easybuild_configuration': eb_config})
+    init_session_state.update({'module_list': modlist})
+    _log.debug("Initial session state: %s" % init_session_state)
+
+    if options.skip_test_step:
+        if options.ignore_test_failure:
+            raise EasyBuildError("Found both ignore-test-failure and skip-test-step enabled. "
+                                 "Please use only one of them.")
+        else:
+            print_warning("Will not run the test step as requested via skip-test-step. "
+                          "Consider using ignore-test-failure instead and verify the results afterwards")
+
+    # determine easybuild-easyconfigs package install path
+    easyconfigs_pkg_paths = get_paths_for(subdir=EASYCONFIGS_PKG_SUBDIR)
+    if not easyconfigs_pkg_paths:
+        _log.warning("Failed to determine install path for easybuild-easyconfigs package.")
+
+    if options.install_latest_eb_release:
```

Comment:
this block should be moved into `rest_of_main` (since it uses `orig_paths`)

Reply:
Done